include/spdlog/details/pattern_formatter_impl.h

```diff
@@ -114,17 +121,15 @@ class B_formatter : public flag_formatter
 };
 
 // write 2 ints separated by sep with padding of 2
-static fmt::MemoryWriter &pad_n_join(fmt::MemoryWriter &w, int v1, int v2, char sep)
+static fmt::memory_buffer &pad_n_join(fmt::memory_buffer &w, int v1, int v2, char sep)
 {
-    w << fmt::pad(v1, 2, '0') << sep << fmt::pad(v2, 2, '0');
-    return w;
+    return w << fmt::format("{:02}", v1) << sep << fmt::format("{:02}", v2);
 }
 
 // write 3 ints separated by sep with padding of 2
-static fmt::MemoryWriter &pad_n_join(fmt::MemoryWriter &w, int v1, int v2, int v3, char sep)
+static fmt::memory_buffer &pad_n_join(fmt::memory_buffer &w, int v1, int v2, int v3, char sep)
 {
-    w << fmt::pad(v1, 2, '0') << sep << fmt::pad(v2, 2, '0') << sep << fmt::pad(v3, 2, '0');
-    return w;
+    return w << fmt::format("{:02}", v1) << sep << fmt::format("{:02}", v2) << sep << fmt::format("{:02}", v3);
```

Comment:
So fmt removed `::pad`, but `::format` should now be of equal performance (if it wasn't before), at least with precompilation.

Reply:
This is not very efficient because every call to `fmt::format` will construct a string. I suggest merging into one call `fmt::format("{1:02}{0}{2:02}{0}{3:02}", sep, v1, v2, v3)`.